draft-ietf-acme-dns-account-01.mkd

```diff
@@ -182,37 +283,32 @@ If the server is unable to find a `TXT` record for the validation domain name, i
-
 # Security Considerations
 
-As this challenge that is introduced only differs in the left-most label of the domain name from the existing `dns-01` challenge, the same security considerations apply.
-
-In terms of the construction of the label prepended to the domain name, there is no need for a cryptographic hash. The purpose of that is to create a long-lived and statistically distinctive record of minimal size.
+These challenges follows the recommendations set out in "Domain Control Validation using DNS" {{I-D.draft-ietf-dnsop-domain-verification-techniques}} for supporting multiple intermediates within the context of {{!RFC8555}}.
 
-SHA-256 was picked due to its broad adoption, hardware support, and existing need in implementations that would likely support `dns-account-01`.
+In both `dns-account-01` and `dns-02`, the same security considerations apply for the integrity of authorizations ({{!RFC8555, Section 10.2}}) and DNS security ({{!RFC8555, Section 11.2}}) as in the original specification for `dns-01`. They both differ from `dns-01` in that the challenges are scoped to the particular name being validated without relying upon CAA ({{!RFC8659}}).
 
-The first 10 bytes were picked as a tradeoff: the value needs to be short enough to not significantly impact DNS record and response size, long enough to provide sufficient probability of collision avoidance across ACME accounts, and just the right size to have Base32 require no padding. As the algorithm is used for uniform distribution of inputs, and not for integrity, we do not consider the trimming a security issue.
-
-# IANA Considerations
+## DNS-ACCOUNT-01
 
-## DNS Parameters
+To allow for seamless account key rollover without the label changing, the dynamic part of the label depends on the ACME account and not the account key. This allows for long-lived labels, without the security considerations of keeping the account key static.
 
-The Underscored and Globally Scoped DNS Node Names is to be updated to include the following entry:
+In terms of the construction of the label prepended to the domain name, there is no need for a cryptographic hash. The purpose of that is to create a long-lived and statistically distinctive record of minimal size. SHA-256 was picked due to it already being used for existing `dns-01` challenges ({{!RFC8555, Section 8.1}}).
 
-~~~
-RR Type: TXT
-_NODE NAME: _acme-challenge_*
-Reference: This document
-~~~
+The first 10 bytes were picked as a tradeoff: the value needs to be short enough to not significantly impact DNS record and response size, long enough to provide sufficient probability of collision avoidance across ACME accounts, and just the right size to have Base32 require no padding. As the algorithm is used for a uniform distribution of inputs, and not for integrity, we do not consider the trimming a security issue.
```

Comment:
The main length concern was to make sure we didn't place additional pressure on the max length of DNS names (253) and make the challenge unusable for abnormally large names. (Statistically there are very very few when I ran this against Censys many years ago.)

I don't think we were too concerned about record size or response size, but it is fine to keep that in.

Reply:
I've clarified this a bit now

Reply:
The max valid length of a DNS name is 253 characters. The max label length is 63. So the full sha256 digest could fit into a base32 encoded label, but it would make more DNS names theoretically not be able to use this challenge.
https://devblogs.microsoft.com/oldnewthing/20120412-00/?p=7873

Certainly smaller responses in general are always preferred to avoid issues with fragmentation attacks.
https://www.ietf.org/archive/id/draft-ietf-dnsop-avoid-fragmentation-16.html